server/sonar-server/src/main/java/org/sonar/server/qualityprofile/QProfileBackuperImpl.java

```diff
@@ -0,0 +1,215 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.server.qualityprofile;
+
+import com.google.common.base.Joiner;
+import com.google.common.base.MoreObjects;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import com.google.common.collect.Sets;
+import java.io.Reader;
+import java.io.Writer;
+import java.util.Comparator;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import javax.annotation.Nullable;
+import javax.xml.stream.XMLInputFactory;
+import javax.xml.stream.XMLStreamException;
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.lang.builder.CompareToBuilder;
+import org.codehaus.staxmate.SMInputFactory;
+import org.codehaus.staxmate.in.SMHierarchicCursor;
+import org.codehaus.staxmate.in.SMInputCursor;
+import org.sonar.api.rule.RuleKey;
+import org.sonar.api.server.ServerSide;
+import org.sonar.api.utils.text.XmlWriter;
+import org.sonar.db.DbClient;
+import org.sonar.db.DbSession;
+import org.sonar.db.organization.OrganizationDto;
+import org.sonar.db.qualityprofile.ActiveRuleDto;
+import org.sonar.db.qualityprofile.ActiveRuleParamDto;
+import org.sonar.db.qualityprofile.QualityProfileDto;
+
+@ServerSide
+public class QProfileBackuperImpl implements QProfileBackuper {
+
+  private static final Joiner RULE_KEY_JOINER = Joiner.on(", ").skipNulls();
+
+  private final QProfileReset reset;
+  private final DbClient db;
+
+  public QProfileBackuperImpl(QProfileReset reset, DbClient db) {
+    this.reset = reset;
+    this.db = db;
+  }
+
+  @Override
+  public void backup(DbSession dbSession, QualityProfileDto profileDto, Writer writer) {
+    List<ActiveRuleDto> activeRules = db.activeRuleDao().selectByProfileKey(dbSession, profileDto.getKey());
+    activeRules.sort(BackupActiveRuleComparator.INSTANCE);
+    writeXml(dbSession, writer, profileDto, activeRules.iterator());
+  }
+
+  private void writeXml(DbSession dbSession, Writer writer, QualityProfileDto profile, Iterator<ActiveRuleDto> activeRules) {
+    XmlWriter xml = XmlWriter.of(writer).declaration();
+    xml.begin("profile");
+    xml.prop("name", profile.getName());
+    xml.prop("language", profile.getLanguage());
+    xml.begin("rules");
+    while (activeRules.hasNext()) {
+      ActiveRuleDto activeRule = activeRules.next();
+      xml.begin("rule");
+      xml.prop("repositoryKey", activeRule.getKey().ruleKey().repository());
+      xml.prop("key", activeRule.getKey().ruleKey().rule());
+      xml.prop("priority", activeRule.getSeverityString());
+      xml.begin("parameters");
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "parameters" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1192)